File: server/src/main/java/com/hospitaltender/server/service/InspectionService.java
package com.hospitaltender.server.service;

import com.hospitaltender.server.dto.request.CreateInspectionRequest;
import com.hospitaltender.server.dto.request.UpdateInspectionRequest;
import com.hospitaltender.server.dto.response.InspectionResponse;
import com.hospitaltender.server.enums.InspectionStatus;

import java.util.List;

public interface InspectionService {

    InspectionResponse create(CreateInspectionRequest request);

    InspectionResponse getById(Long id);

    List<InspectionResponse> getByTenderId(Long tenderId);

    List<InspectionResponse> getByStatus(InspectionStatus status);

    InspectionResponse update(Long id, UpdateInspectionRequest request);

    InspectionResponse updateStatus(Long id, InspectionStatus status);

    void delete(Long id);
}

File: ui/hospital-system-ui/src/services/inspection.service.ts
import api from './api'
import {
  InspectionResponse,
  CreateInspectionRequest,
  UpdateInspectionRequest,
  InspectionStatus,
} from '@/types'

export const inspectionService = {
  getByTenderId: (tenderId: number) =>
    api.get<InspectionResponse[]>(`/inspections/tender/${tenderId}`).then((res) => res.data),

  getById: (id: number) =>
    api.get<InspectionResponse>(`/inspections/${id}`).then((res) => res.data),

  getByStatus: (status: InspectionStatus) =>
    api.get<InspectionResponse[]>(`/inspections/status/${status}`).then((res) => res.data),

  create: (data: CreateInspectionRequest) =>
    api.post<InspectionResponse>('/inspections', data).then((res) => res.data),

  update: (id: number, data: UpdateInspectionRequest) =>
    api.put<InspectionResponse>(`/inspections/${id}`, data).then((res) => res.data),

  updateStatus: (id: number, status: InspectionStatus) =>
    api.patch<InspectionResponse>(`/inspections/${id}/status`, null, { params: { status } }).then((res) => res.data),

  delete: (id: number) =>
    api.delete(`/inspections/${id}`),
}
